src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -585,6 +586,7 @@ private void IdentifyUserFunctions(ParseNode node)
                 context.UserFunctions.GetUserFunction(funcIdentifier, storageType == StorageModifier.GLOBAL ? (Int16)0 : GetContainingScopeId(node), node);
             int expressionHash = ConcatenateNodes(bodyNode).GetHashCode();
             userFuncObject.GetUserFunctionOpcodes(expressionHash);
+            userFuncObject.IsFunction = !isLock;
```

Comment:
Is it possible for userFuncObject to already be IsFunction before this is called, but not be a lock (i.e. it used `define function....`), in which case this line of code would turn its flag *off* when it had previously been on.

In any case, I'd be more comfortable with a one-way flip here:

```
if (isLock)
    userFuncObject.IsFunction = true;
```

That way this line can make the flag become true, but can't make it become false if it was already true for other reasons.